ts41/camel-cased-props.d.ts

```diff
@@ -0,0 +1,28 @@
+import {CamelCase} from './camel-case';
+
+/**
+Convert object props to camelCase but not recursively.
+
+This can be useful when, for example, converting some api type from other style.
+
+@see CamelCasedPropsDeep
+@see CamelCase
+@example
+```
+interface User {
+	UserId: number;
+	UserName: string;
+}
+
+const result: CamelCasedProps<User> = {
+	userId: 1,
+	userName: "Tom",
+};
+
+```
+*/
+export type CamelCasedProps<T> = T extends Array<infer U>
+	? U[]
```

Comment:
Why not return `T` in this case? Could potentially be a tuple and `U[]` would turn it into an array unnecessarily.

Not sure if it makes sense to accept an array here at all though, might be better to restrict it to objects keyed with strings.

Reply:
resolved, prefer to return just T instead of restrict it, cause return just T make it have similar behaviour as `Deep` version